src/headers.rs

```diff
@@ -56,6 +57,34 @@ impl<'a> IntoIterator for Headers<'a> {
     }
 }
 
+/// Allows formatting and printing the `Headers` struct items.
+///
+/// # Examples
+/// ```
+///     use mailparse::parse_mail;
+///     let mail = parse_mail(concat!(
+///             "Subject: foo\n",
+///             "Another header: bar\n",
+///             "\n",
+///             "Body starts here").as_bytes())
+///         .unwrap();
+///     let mut headers = mail.get_headers();
+///     assert_eq!(format!("{:?}", headers), "Headers { raw_bytes: \
+///                [83, 117, 98, 106, 101, 99, 116, 58, 32, 102, 111, 111, 10, \
+///                65, 110, 111, 116, 104, 101, 114, 32, 104, 101, 97, 100, 101, \
+///                114, 58, 32, 98, 97, 114, 10, 10], \
+///                headers: [MailHeader { key: \"Subject\", value: \"foo\" }, \
+///                MailHeader { key: \"Another header\", value: \"bar\" }] }");
+/// ```
+impl<'a> fmt::Debug for Headers<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Headers")
+            .field("raw_bytes", &self.raw_bytes)
```

Comment:
I'm not sure if it's useful to dump the raw_bytes here at all. I think just dumping the `self.headers` part should be good - but I'm open to being convinced otherwise if you feel that dumping the raw bytes is useful. For now I'll mark this PR as changes requested, but if you feel the raw_bytes is useful please let me know what the circumstances are.

Other than that this patch looks fine to me, thanks!

Reply:
OK. We use raw bytes to debug our protocol parsers but if it does not make sense for most of your users then it should probably not be there. I think we can get raw bytes other ways too.
I have a question though. If we just print certain elements of a struct, would it not give an impression that the struct only has those elements? I'm still learning Rust, I do not know if it is common to let certain fields of a struct not be dumped when struct is printed.

Reply:
In general I don't think people should be looking at the debug formatter output to figure out which elements are in a struct. It's better to look at the documentation or source code for that.

In this case, the raw_bytes and headers really represent the same underlying information; the headers is a thin wrapper over top of the raw bytes, and arguably the one that is more useful for humans looking at the debug output. And anyway whoever called `parse_mail` already has the raw bytes so they can examine those outside of `mailparse` too. If anything, it might make sense to augment the debug output with other kinds of information that is specific to the parsing process - for example, indices of line breaks or header key/value delimiters.

Reply:
Thank you for the explanation!